launcher/mainwindow_moc.cpp

```diff
@@ -236,16 +236,32 @@ void MainWindow::on_aboutButton_clicked()
 void MainWindow::updateTranslation()
 {
 #ifdef ENABLE_QT_TRANSLATIONS
-	const std::string translationFile = settings["general"]["language"].String() + ".qm";
-	logGlobal->info("Loading translation '%s'", translationFile);
+	const std::string translationFile = settings["general"]["language"].String()+ ".qm";
+	QString translationFileResourcePath = QString{":/translation/%1"}.arg(translationFile.c_str());
 
-	if (!translator.load(QString{":/translation/%1"}.arg(translationFile.c_str())))
+	logGlobal->info("Loading translation %s", translationFile);
+
+	if(!QFile::exists(translationFileResourcePath))
+	{
+		logGlobal->warn("Translation file %s does not exist", translationFileResourcePath.toStdString());
+		return;
+	}
+
+	if (!translator.load(translationFileResourcePath))
 	{
-		logGlobal->error("Failed to load translation");
+		logGlobal->error("Failed to load translation file %s", translationFileResourcePath.toStdString());
+		return;
+	}
+
+	if(translationFile == "english.qm")
+	{
+		// translator doesn't need to be installed for English
```

Comment:
Won't this result in translation not changing to English if I were to switch from language that has translation to English?

Reply:
No, the translation does switch back to English when selects the following sequence:

English -> German -> English

It does not work if I put the check before `translator.load`